src/hooks/auditoriumSections.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useCallback, useMemo } from "react";
+
+import { UserProfilePicture } from "components/molecules/UserProfilePicture";
+
+import { setGridData } from "api/profile";
+
+import { WithId } from "utils/id";
+import { convertCoordinate } from "utils/auditorium";
+
+import { GridPosition } from "types/grid";
+import { User } from "types/User";
+
+import {
+  currentAuditoriumSectionsSelector,
+  currentAuditoriumSectionsByIdSelector,
+} from "utils/selectors";
+
+import { useSelector } from "./useSelector";
+import { useFirestoreConnect, isLoaded } from "./useFirestoreConnect";
+import { useRecentVenueUsers } from "./users";
+import { useUser } from "./useUser";
+import { useGetUserByPosition } from "./useGetUserByPosition";
+
+const useConnectAuditoriumSections = (venueId?: string) => {
+  useFirestoreConnect(() => {
+    if (!venueId) return [];
+
+    return [
+      {
+        collection: "venues",
+        doc: venueId,
+        subcollections: [{ collection: "sections" }],
+        storeAs: "currentAuditoriumSections",
+      },
+    ];
+  });
+};
+
+export interface UseAuditoriumSectionProps {
+  rows: number;
+  columns: number;
+  videoWidthInSeats: number;
+  videoHeightInSeats: number;
+  sectionId?: string;
+  venueId?: string;
+}
+
+export const useAuditoriumSection = ({
+  venueId,
+  sectionId,
+  rows,
+  columns,
+  videoWidthInSeats,
+  videoHeightInSeats,
+}: UseAuditoriumSectionProps) => {
+  useConnectAuditoriumSections(venueId);
+
+  const { userWithId } = useUser();
+  const userId = userWithId?.id;
+
+  const sectionsById = useSelector(currentAuditoriumSectionsByIdSelector);
+  const section = sectionId ? sectionsById?.[sectionId] : undefined;
+
+  const seatedUsers = useSectionSeatedUsers(venueId, sectionId);
+
+  const getUserBySeat = useGetUserByPosition({
+    venueId,
+    positionedUsers: seatedUsers,
+  });
+
+  const takeSeat = useCallback(
+    ({ row, column }: GridPosition) => {
+      if (!sectionId || !venueId || !userId) return;
+
+      setGridData({
+        venueId,
+        userId,
+        gridData: { sectionId, row, column },
+      });
+    },
+    [sectionId, venueId, userId]
+  );
```

Comment:
We should return the value of `setGridData` so that it 'passes on' it's `Promise`. It would be good to set the type of `takeSeat` explicitly to make it more obvious what it's intended to be here (`Promise<void>`)